### PR TITLE
cmd/k8s-operator: helm chart add missing keys

### PR DESCRIPTION
**Pull request by GMartinez-Sisti**
Closes https://github.com/tailscale/tailscale/issues/10182.

Add missing keys to helm chart default values.

### CODE REVIEW

**Comment by PaulSonOfLars** [comment]
I believe this is missing the  `.Values.oauth` fields (in the secret template).

(apologies if this next bit is out of scope for this PR - just a passing thought)
Would it be worth using an boolean to enable/disable secrets? This would allow for demonstrating that the field exists, without it being set.

eg:

```
{{ if and .Values.oauth .Values.oauth.create -}}
apiVersion: v1
kind: Secret
metadata:
  name: operator-oauth
  namespace: {{ .Release.Namespace }}
stringData:
  client_id: {{ .Values.oauth.clientId }}
  client_secret: {{ .Values.oauth.clientSecret }}
{{- end -}}
```

**Comment by GMartinez-Sisti** [comment]
> I believe this is missing the `.Values.oauth` fields (in the secret template).
> 
> (apologies if this next bit is out of scope for this PR - just a passing thought. Happy to open a PR for it if requested) Would it be worth using a boolean to enable/disable secrets? This would allow for demonstrating that the field exists, without it being set.
> 
> eg:
> 
> ```
> {{ if and .Values.oauth .Values.oauth.create -}}
> apiVersion: v1
> kind: Secret
> metadata:
>   name: operator-oauth
>   namespace: {{ .Release.Namespace }}
> stringData:
>   client_id: {{ .Values.oauth.clientId }}
>   client_secret: {{ .Values.oauth.clientSecret }}
> {{- end -}}
> ```

Hi @PaulSonOfLars! Those values are commented on the first lines of the values chart:

https://github.com/tailscale/tailscale/blob/456c9c4d3aacfba8e883377b2f5e9f97c53d1c65/cmd/k8s-operator/deploy/chart/values.yaml#L4-L9

I didn't uncomment or the current logic would be broken:

https://github.com/tailscale/tailscale/blob/456c9c4d3aacfba8e883377b2f5e9f97c53d1c65/cmd/k8s-operator/deploy/chart/templates/oauth-secret.yaml#L4

I like your approach but that would be a breaking change. Waiting for the maintainers opinion.

**Review by irbekrm — CHANGES_REQUESTED** [review]
Thank you for the PR @GMartinez-Sisti !

Looks good to me except for the resource values- I left a comment there.

**Comment by irbekrm** [comment]
>Hi @PaulSonOfLars! Those values are commented on the first lines of the values chart:

Looks like the oauth block in `values.yaml` is easy to miss as it's all commented out and blends in with the licence header above.
We could uncomment line 7 and make it `oauth: {}` that will work with the current syntax as [helm considers an empty map a nil value](https://helm.sh/docs/chart_template_guide/control_structures/#ifelse) - and should hopefully make it more noticeable.

File: cmd/k8s-operator/deploy/chart/values.yaml [hunk 1]
```diff
@@ -15,11 +15,29 @@ operatorConfig:
...
   nodeSelector:
     kubernetes.io/os: linux
 
+  resources: {}
+  # limits:
+  #   cpu: 100m
+  #   memory: 128Mi
+  # requests:
+  #   cpu: 100m
+  #   memory: 128Mi
```
[the remark this answers is not in the document]
Comment (GMartinez-Sisti) [reply cmd/k8s-operator/deploy/chart/values.yaml]: I agree. Removed.

**Review by irbekrm — APPROVED** [review]
Thank you for the PR and for making the change!

**Pull request merged**